contrib/src/main/java/com/datatorrent/contrib/parser/LogSchemaDetails.java

```diff
@@ -0,0 +1,230 @@
+      this.fields.add(field);
+      this.fieldNames.add(field.name);
+    }
+  }
+
+  /**
+   * creates regex group pattern from the regex given for each field
+   */
+  public void createPattern()
+  {
+    StringBuffer pattern = new StringBuffer();
+    for(Field field: this.getFields()) {
+      pattern.append(field.getRegex()).append(" ");
+    }
+    logger.info("Created pattern for parsing the log {}", pattern.toString().trim());
+    this.setPattern(pattern.toString().trim());
+  }
+
+  /**
+   * creates json object by matching the log with given pattern
+   * @param log
+   * @return logObject
+   * @throws Exception
+   */
+  public JSONObject createJsonFromLog(String log) throws JSONException
+  {
+    Pattern compile = Pattern.compile(this.pattern);
+    Matcher m = compile.matcher(log);
+    int count = m.groupCount();
+    JSONObject logObject = null;
+    if (m.find()) {
+      int i = 1;
+      logObject  = new JSONObject();
+      for(String field: this.getFieldNames()) {
+        if (i > count) {
+          break;
+        }
+        logObject.put(field, m.group(i));
+        i++;
+      }
+    }
+    return logObject;
+  }
+
+  /**
+   * Get the list of fieldNames mentioned in schema
+   * @return fieldNames
+   */
+  public List<String> getFieldNames()
+  {
+    return fieldNames;
+  }
+
+  /**
+   * Get the list of fields (field, regex) mentioned in schema
+   * @return fields
+   */
+  public List<Field> getFields()
+  {
+    return fields;
+  }
+
+  /**
+   * Get the regex pattern for the schema
+   * @return pattern
+   */
+  public String getPattern()
+  {
+    return pattern;
+  }
+
+  /**
+   * Set the regex pattern for schema
+   * @param pattern
+   */
+  public void setPattern(String pattern)
+  {
+    this.pattern = pattern;
+  }
+
+  public class Field
+  {
+    /**
+     * name of the field
+     */
+    private String name;
+    /**
+     * regular expression for the field
+     */
+    private String regex;
+
+
+    public Field(String name, String regex)
+    {
+      this.name = name;
+      this.regex = regex;
+    }
+
+    /**
+     * Get the name of the field
+     * @return name
+     */
+    public String getName()
+    {
+      return name;
+    }
+
+    /**
+     * Set the name of the field
+     * @param name
+     */
+    public void setName(String name)
+    {
+      this.name = name;
+    }
+
+    /**
+     * Get the regular expression of the field
+     * @return regex
+     */
+    public String getRegex()
+    {
+      return regex;
+    }
+
+    /**
+     * Set the regular expression of the field
+     * @param regex
+     */
+    public void setRegex(String regex)
+    {
+      this.regex = regex;
+    }
+
+    @Override
+    public String toString()
+    {
+      return "Fields [name=" + name + ", regex=" + regex +"]";
+    }
+  }
+
+  private static final Logger logger = LoggerFactory.getLogger(LogSchemaDetails.class);
+}
```

Comment:
@tushargosavi  I have done all the changes suggested by you. Missing new line at the end is also done in this as well as previous commit.